### ISSUE TITLE
installation problems mamba/micromamba

### ISSUE THREAD

**Issue by sebbruchmann** [kelwyres/Bactabolize#30]
Hi guys,
I tried to install bactabolize on our unix high performance cluster but ran into some installation issues.
Because of the restrictive rights on the cluster, I have limited options to install software, conda/mamba/micromamba is usually my preferred choice.
I have tried installing it with mamba and micromamba but both failed. Below are some more details of what I have done and what the error messages are.
Happy to provide more details if needed. Do you have an idea where I went wrong?

Many thanks,
Seb

```
micromamba create -n bactabolize
micromamba activate bactabolize
```

> micromamba install -c kelwyres -c bioconda -c conda-forge 'bactabolize==1.0.3' -y
> error    libmamba Could not solve for environment specs
>     The following package could not be installed
>     └─ bactabolize ==1.0.3 * is not installable because it requires
>        └─ biopython ==1.79 *, which conflicts with any installable versions previously reported.

next try, from fresh environment:

```
micromamba install -c conda-forge "biopython==1.79" -y
micromamba install -c kelwyres -c bioconda -c conda-forge 'bactabolize==1.0.3' -y
```

> error    libmamba Could not solve for environment specs
>     The following packages are incompatible
>     ├─ bactabolize 1.0.3  is installable and it requires
>     │  └─ python 3.9 , which can be installed;
>     └─ pin-1 is not installable because it requires
>        └─ python 3.11.* , which conflicts with any installable versions previously reported.

next try, from fresh environment:

```
micromamba install -c conda-forge "python==3.9" -y
micromamba install -c conda-forge "biopython==1.79" -y
micromamba install -c kelwyres -c bioconda -c conda-forge 'bactabolize==1.0.3' -y
```

> error    libmamba Could not solve for environment specs
>     The following packages are incompatible
>     ├─ bactabolize 1.0.3  is installable and it requires
>     │  └─ requests with the potential options
>     │     ├─ requests [2.10.0|2.11.0|...|2.9.2] would require
>     │     │  └─ python 2.7* , which can be installed;
>     │     ├─ requests [2.10.0|2.11.0|...|2.9.2] would require
>     │     │  └─ python 3.4* , which can be installed;
>     │     ├─ requests [2.10.0|2.11.0|...|2.9.2] would require
>     │     │  └─ python 3.5* , which can be installed;
>     │     ├─ requests [2.12.4|2.12.5|2.13.0] would require
>     │     │  └─ python 3.6* , which can be installed;
>     │     ├─ requests [2.17.3|2.18.1|2.18.2|2.18.3|2.18.4] would require
>     │     │  └─ urllib3 >=1.21.1,<1.22 , which conflicts with any installable versions previously reported;
>     │     ├─ requests 2.18.4 would require
>     │     │  └─ urllib3 >=1.21.1,<1.23 , which conflicts with any installable versions previously reported;
>     │     ├─ requests [2.19.0|2.19.1|2.20.0|2.20.1] would require
>     │     │  └─ urllib3 >=1.21.1,<1.24 , which conflicts with any installable versions previously reported;
>     │     ├─ requests [2.21.0|2.22.0] would require
>     │     │  └─ urllib3 >=1.21.1,<1.25 , which conflicts with any installable versions previously reported;
>     │     ├─ requests [2.22.0|2.23.0|2.24.0|2.25.0] would require
>     │     │  └─ urllib3 >=1.21.1,<1.26,!=1.25.0,!=1.25.1 , which conflicts with any installable versions previously reported;
>     │     ├─ requests [2.25.1|2.26.0|...|2.29.0] would require
>     │     │  └─ urllib3 >=1.21.1,<1.27 , which conflicts with any installable versions previously reported;
>     │     ├─ requests [2.31.0|2.32.1|2.32.2|2.32.3] would require
>     │     │  └─ urllib3 >=1.21.1,<3 , which conflicts with any installable versions previously reported;
>     │     └─ requests [2.18.4|2.19.1|...|2.32.3] conflicts with any installable versions previously reported;
>     └─ pin-1 is not installable because it requires
>        └─ python 3.9.* , which conflicts with any installable versions previously reported.

`micromamba install -c conda-forge urllib3 -y`

worked
run install again:

`micromamba install -c kelwyres -c bioconda -c conda-forge 'bactabolize==1.0.3' -y
`

> error    libmamba Could not solve for environment specs
>     The following package could not be installed
>     └─ bactabolize 1.0.3  is not installable because it requires
>        └─ cobra 0.21.0  but there are no viable options
>           ├─ cobra 0.21.0 would require
>           │  └─ swiglpk >=4.65  but there are no viable options
>           │     ├─ swiglpk [4.65.0|4.65.1|...|5.0.5] would require
>           │     │  └─ glpk >=4.65,<4.66.0a0 , which conflicts with any installable versions previously reported;
>           │     └─ swiglpk [5.0.10|5.0.11|...|5.0.9] would require
>           │        └─ glpk >=5.0,<6.0a0 , which conflicts with any installable versions previously reported;
>           └─ cobra 0.21.0 conflicts with any installable versions previously reported.

`micromamba install -c conda-forge glpk -y`
run install again:
`micromamba install -c kelwyres -c bioconda -c conda-forge 'bactabolize==1.0.3' -y`

> error    libmamba Could not solve for environment specs
>     The following packages are incompatible
>     ├─ bactabolize 1.0.3  is installable and it requires
>     │  └─ goodtables ~=2.0  with the potential options
>     │     ├─ goodtables [2.1.6|2.2.0|...|2.4.2] would require
>     │     │  └─ python >=2.7,<2.8.0a0 , which can be installed;
>     │     ├─ goodtables [2.1.6|2.2.0|...|2.5.4] would require
>     │     │  └─ python >=3.6,<3.7.0a0 , which can be installed;
>     │     ├─ goodtables [2.1.6|2.2.0|...|2.5.4] would require
>     │     │  └─ python >=3.7,<3.8.0a0 , which can be installed;
>     │     ├─ goodtables [2.4.0|2.4.1|2.4.2|2.5.2|2.5.4] would require
>     │     │  └─ python >=3.8,<3.9.0a0 , which can be installed;
>     │     ├─ goodtables [2.5.2|2.5.4] would require
>     │     │  └─ tabulator >=1.20  with the potential options
>     │     │     ├─ tabulator [1.20.0|1.21.0] would require
>     │     │     │  └─ cchardet >=1.0,<2.0  with the potential options
>     │     │     │     ├─ cchardet 1.1.3 would require
>     │     │     │     │  └─ python 2.7* , which can be installed;
>     │     │     │     ├─ cchardet 1.1.3 would require
>     │     │     │     │  └─ python 3.5* , which can be installed;
>     │     │     │     └─ cchardet 1.1.3 would require
>     │     │     │        └─ python 3.6* , which can be installed;
>     │     │     ├─ tabulator [1.21.0|1.22.0|...|1.25.1] would require
>     │     │     │  └─ openpyxl >=2.4,<2.5  with the potential options
>     │     │     │     ├─ openpyxl 2.4.0 would require
>     │     │     │     │  └─ python 2.7* , which can be installed;
>     │     │     │     ├─ openpyxl 2.4.0 would require
>     │     │     │     │  └─ python 3.4* , which can be installed;
>     │     │     │     ├─ openpyxl 2.4.0 would require
>     │     │     │     │  └─ python 3.5* , which can be installed;
>     │     │     │     ├─ openpyxl 2.4.0 would require
>     │     │     │     │  └─ python 3.6* , which can be installed;
>     │     │     │     └─ openpyxl [2.4.10|2.4.11|...|2.5.0b1] conflicts with any installable versions previously reported;
>     │     │     └─ tabulator [1.26.1|1.27.0|...|1.53.5] would require
>     │     │        └─ openpyxl >=2.6 , which conflicts with any installable versions previously reported;
>     │     └─ goodtables 2.5.4 would require
>     │        └─ tabulator >=1.40 , which cannot be installed (as previously explained);
>     └─ pin-1 is not installable because it requires
>        └─ python 3.9.* , which conflicts with any installable versions previously reported.
> critical libmamba Could not solve for environment specs

this is where I gave up ...

**Comment by bananabenana** [comment]
Hi @sebbruchmann ,

This is frustrating, I totally get ya, and I'm sorry the installation has been difficult thus far.

I can't test this myself as this is specific to your computing environment. I would just confirm that you're in your new correct environment when installing all these and not just modifying base environment (which it sounds like you are already doing). 

I think you're close though, if you keep installing the deps that pop up. These are deps of deps (dep-deps), so to ensure correct versioning, the block at the end of this comment is the result of `conda list` within a fresh Bactabolize env which lists specific versions and everything required. One issue is, you may be installing recent versions of dep-deps, which have newer requirements. So hopefully using this `conda list` will anchor the deps in a compatible way.

Try to see if the required deps are installed in the correct version, so your installation commands may look as follows:
```bash
micromamba install -y -c conda-forge python==3.9
micromamba install -y -c conda-forge biopython==1.79
micromamba install -y -c conda-forge glpk==5.0
micromamba install -y -c conda-forge swiglpk==5.0.10
micromamba install -y -c conda-forge urllib3=2.2.2
micromamba install -y -c conda-forge cchardet==2.1.7
micromamba install -y -c conda-forge tabulator=1.25.1
micromamba install -y -c conda-forge goodtables==2.5.4
micromamba install -y -c conda-forge openpyxl==2.4.11
micromamba install -y -c kelwyres -c bioconda -c conda-forge bactabolize==1.0.3
```

conda list output of a functioning Bactabolize env (tab-delimited):
```
Name	Version	Build	Channel
_libgcc_mutex	0.1	conda_forge	conda-forge
_openmp_mutex	4.5	2_gnu	conda-forge
alabaster	0.7.16	pyhd8ed1ab_0	conda-forge
annotated-types	0.7.0	pyhd8ed1ab_0	conda-forge
anyio	4.4.0	pyhd8ed1ab_0	conda-forge
appdirs	1.4.4	pyh9f0ad1d_0	conda-forge
arrow	1.3.0	pyhd8ed1ab_0	conda-forge
attrs	23.2.0	pyh71513ae_0	conda-forge
babel	2.14.0	pyhd8ed1ab_0	conda-forge
bactabolize	1.0.3	pypi_0	pypi
binaryornot	0.4.4	py_1	conda-forge
biopython	1.79	py39hb9d737c_3	conda-forge
blast	2.12.0	hf3cf87c_4	bioconda
brotli-python	1.1.0	py39h3d6467e_1	conda-forge
bzip2	1.0.8	hd590300_5	conda-forge
c-ares	1.32.0	h4ab18f5_0	conda-forge
ca-certificates	2024.7.4	hbcca054_0	conda-forge
cchardet	2.1.7	py39h3d6467e_5	conda-forge
certifi	2024.7.4	pyhd8ed1ab_0	conda-forge
cffi	1.15.1	py39h74dc2b5_0
chardet	5.2.0	py39hf3d152e_1	conda-forge
charset-normalizer	3.3.2	pyhd8ed1ab_0	conda-forge
click	8.1.7	unix_pyh707e725_0	conda-forge
click-config-file	0.6.0	pyhd8ed1ab_1	conda-forge
click-default-group	1.2.4	pyhd8ed1ab_0	conda-forge
click-log	0.4.0	pyhd8ed1ab_0	conda-forge
cobra	0.21.0	pyh5e36f6f_1	bioconda
colorama	0.4.6	pyhd8ed1ab_0	conda-forge
configobj	5.0.8	pyhd8ed1ab_0	conda-forge
cookiecutter	2.6.0	pyhca7485f_0	conda-forge
curl	7.88.1	h37d81fd_2
datapackage	1.15.2	pyh44b312d_0	conda-forge
depinfo	1.7.0	pyhd8ed1ab_0	conda-forge
diskcache	5.6.3	pyhd8ed1ab_0	conda-forge
docutils	0.21.2	pyhd8ed1ab_0	conda-forge
entrez-direct	22.1	he881be0_0	bioconda
et_xmlfile	1.1.0	pyhd8ed1ab_0	conda-forge
exceptiongroup	1.2.0	pyhd8ed1ab_2	conda-forge
filelock	3.8.0	pyhd8ed1ab_0	conda-forge
future	1.0.0	pyhd8ed1ab_0	conda-forge
gettext	0.22.5	h59595ed_2	conda-forge
gettext-tools	0.22.5	h59595ed_2	conda-forge
gitdb	4.0.11	pyhd8ed1ab_0	conda-forge
gitpython	3.1.43	pyhd8ed1ab_0	conda-forge
glpk	5.0	h445213a_0	conda-forge
gmp	6.3.0	hac33072_2	conda-forge
gmpy2	2.1.5	py39h048c657_1	conda-forge
goodtables	2.5.4	py39hf3d152e_0	conda-forge
greenlet	3.0.3	py39h3d6467e_0	conda-forge
h11	0.14.0	pyhd8ed1ab_0	conda-forge
h2	4.1.0	py39hf3d152e_0	conda-forge
hpack	4.0.0	pyh9f0ad1d_0	conda-forge
httpcore	1.0.5	pyhd8ed1ab_0	conda-forge
httpx	0.27.0	pyhd8ed1ab_0	conda-forge
hyperframe	6.0.1	pyhd8ed1ab_0	conda-forge
idna	3.7	pyhd8ed1ab_0	conda-forge
ijson	3.3.0	pyhd8ed1ab_0	conda-forge
imagesize	1.4.1	pyhd8ed1ab_0	conda-forge
importlib-metadata	8.0.0	pyha770c72_0	conda-forge
importlib_metadata	8.0.0	hd8ed1ab_0	conda-forge
importlib_resources	5.12.0	pyhd8ed1ab_0	conda-forge
iniconfig	2.0.0	pyhd8ed1ab_0	conda-forge
isodate	0.6.1	pyhd8ed1ab_0	conda-forge
jdcal	1.4.1	py_0	conda-forge
jinja2	3.1.4	pyhd8ed1ab_0	conda-forge
jsonlines	4.0.0	pyhd8ed1ab_0	conda-forge
jsonpointer	3.0.0	py39hf3d152e_0	conda-forge
jsonschema	4.23.0	pyhd8ed1ab_0	conda-forge
jsonschema-specifications	2023.12.1	pyhd8ed1ab_0	conda-forge
keyutils	1.6.1	h166bdaf_0	conda-forge
krb5	1.20.1	hf9c8cef_0	conda-forge
ld_impl_linux-64	2.40	hf3520f5_7	conda-forge
libasprintf	0.22.5	h661eb56_2	conda-forge
libasprintf-devel	0.22.5	h661eb56_2	conda-forge
libblas	3.9.0	22_linux64_openblas	conda-forge
libcblas	3.9.0	22_linux64_openblas	conda-forge
libcurl	7.88.1	h91b91d3_2
libedit	3.1.20230828	h5eee18b_0
libev	4.33	hd590300_2	conda-forge
libffi	3.3	h58526e2_2	conda-forge
libgcc-ng	14.1.0	h77fa898_0	conda-forge
libgettextpo	0.22.5	h59595ed_2	conda-forge
libgettextpo-devel	0.22.5	h59595ed_2	conda-forge
libgfortran-ng	14.1.0	h69a702a_0	conda-forge
libgfortran5	14.1.0	hc5f4f2c_0	conda-forge
libgomp	14.1.0	h77fa898_0	conda-forge
libidn2	2.3.7	hd590300_0	conda-forge
liblapack	3.9.0	22_linux64_openblas	conda-forge
libnghttp2	1.52.0	ha637b67_1
libopenblas	0.3.27	pthreads_hac2b453_1	conda-forge
libsqlite	3.46.0	hde9e2c9_0	conda-forge
libssh2	1.10.0	haa6b8db_3	conda-forge
libstdcxx-ng	14.1.0	hc0a3c3a_0	conda-forge
libunistring	0.9.10	h7f98852_0	conda-forge
libxcrypt	4.4.36	hd590300_1	conda-forge
libzlib	1.2.13	h4ab18f5_6	conda-forge
linear-tsv	1.1.0	py_1	conda-forge
markdown-it-py	3.0.0	pyhd8ed1ab_0	conda-forge
markupsafe	2.1.5	py39hd1e30aa_0	conda-forge
mdurl	0.1.2	pyhd8ed1ab_0	conda-forge
mpc	1.3.1	hfe3b2da_0	conda-forge
mpfr	4.2.1	h9458935_1	conda-forge
mpmath	1.3.0	pyhd8ed1ab_0	conda-forge
ncurses	6.5	h59595ed_0	conda-forge
numpy	1.23.5	py39h3d75532_0	conda-forge
numpydoc	1.7.0	pyhd8ed1ab_1	conda-forge
openpyxl	2.4.11	py_0	conda-forge
openssl	1.1.1w	hd590300_0	conda-forge
optlang	1.4.4	pyh9f0ad1d_0	conda-forge
packaging	24.1	pyhd8ed1ab_0	conda-forge
pandas	2.2.2	py39hfc16268_1	conda-forge
pcre	8.45	h9c3ff4c_0	conda-forge
perl	5.32.1	7_hd590300_perl5	conda-forge
perl-archive-tar	2.40	pl5321hdfd78af_0	bioconda
perl-carp	1.50	pl5321hd8ed1ab_0	conda-forge
perl-common-sense	3.75	pl5321hdfd78af_0	bioconda
perl-compress-raw-bzip2	2.201	pl5321h87f3376_1	bioconda
perl-compress-raw-zlib	2.202	pl5321h166bdaf_0	conda-forge
perl-encode	3.21	pl5321hd590300_0	conda-forge
perl-exporter	5.74	pl5321hd8ed1ab_0	conda-forge
perl-exporter-tiny	1.002002	pl5321hdfd78af_0	bioconda
perl-extutils-makemaker	7.70	pl5321hd8ed1ab_0	conda-forge
perl-io-compress	2.201	pl5321hdbdd923_2	bioconda
perl-io-zlib	1.14	pl5321hdfd78af_0	bioconda
perl-json	4.10	pl5321hdfd78af_0	bioconda
perl-json-xs	2.34	pl5321h4ac6f70_6	bioconda
perl-list-moreutils	0.430	pl5321hdfd78af_0	bioconda
perl-list-moreutils-xs	0.430	pl5321h031d066_2	bioconda
perl-parent	0.241	pl5321hd8ed1ab_0	conda-forge
perl-pathtools	3.75	pl5321hec16e2b_3	bioconda
perl-scalar-list-utils	1.63	pl5321h166bdaf_0	conda-forge
perl-storable	3.15	pl5321hec16e2b_3	bioconda
perl-types-serialiser	1.01	pl5321hdfd78af_0	bioconda
pip	24.0	pyhd8ed1ab_0	conda-forge
pkgutil-resolve-name	1.3.10	pyhd8ed1ab_1	conda-forge
pluggy	1.5.0	pyhd8ed1ab_0	conda-forge
prodigal	2.6.3	h031d066_9	bioconda
pycparser	2.22	pyhd8ed1ab_0	conda-forge
pydantic	2.8.2	pyhd8ed1ab_0	conda-forge
pydantic-core	2.20.1	py39h5cde264_0	conda-forge
pygments	2.18.0	pyhd8ed1ab_0	conda-forge
pylru	1.2.1	pyhd8ed1ab_0	conda-forge
pysocks	1.7.1	py39hf3d152e_5	conda-forge
pytest	8.2.2	pyhd8ed1ab_0	conda-forge
python	3.9.0	hffdb5ce_5_cpython	conda-forge
python-dateutil	2.9.0post0	py39h06a4308_2
python-libsbml	5.20.2	py39h3d6467e_2	conda-forge
python-slugify	8.0.4	pyhd8ed1ab_0	conda-forge
python-tzdata	2024.1	pyhd8ed1ab_0	conda-forge
python_abi	3.9	4_cp39	conda-forge
pytz	2024.1	pyhd8ed1ab_0	conda-forge
pyyaml	6.0.1	py39hd1e30aa_1	conda-forge
readline	8.2	h8228510_1	conda-forge
referencing	0.35.1	pyhd8ed1ab_0	conda-forge
requests	2.32.3	pyhd8ed1ab_0	conda-forge
rfc3986	2.0.0	pyhd8ed1ab_0	conda-forge
rich	13.7.1	pyhd8ed1ab_0	conda-forge
rpds-py	0.19.0	py39h5cde264_0	conda-forge
ruamel.yaml	0.18.6	py39hd1e30aa_0	conda-forge
ruamel.yaml.clib	0.2.8	py39hd1e30aa_0	conda-forge
setuptools	70.2.0	pyhd8ed1ab_0	conda-forge
simpleeval	0.9.13	py39hd1e30aa_1	conda-forge
six	1.16.0	pyh6c4a22f_0	conda-forge
smmap	5.0.0	pyhd8ed1ab_0	conda-forge
sniffio	1.3.1	pyhd8ed1ab_0	conda-forge
snowballstemmer	2.2.0	pyhd8ed1ab_0	conda-forge
sphinx	7.3.7	pyhd8ed1ab_0	conda-forge
sphinxcontrib-applehelp	1.0.8	pyhd8ed1ab_0	conda-forge
sphinxcontrib-devhelp	1.0.6	pyhd8ed1ab_0	conda-forge
sphinxcontrib-htmlhelp	2.0.5	pyhd8ed1ab_0	conda-forge
sphinxcontrib-jsmath	1.0.1	pyhd8ed1ab_0	conda-forge
sphinxcontrib-qthelp	1.0.7	pyhd8ed1ab_0	conda-forge
sphinxcontrib-serializinghtml	1.1.10	pyhd8ed1ab_0	conda-forge
sqlalchemy	2.0.31	py39hd3abc70_0	conda-forge
sqlite	3.46.0	h6d4b2fc_0	conda-forge
swiglpk	5.0.10	py39hd1e30aa_0	conda-forge
sympy	1.12.1	pypyh2585a3b_103	conda-forge
tableschema	1.17.0	pyh9f0ad1d_0	conda-forge
tabulate	0.9.0	pyhd8ed1ab_1	conda-forge
tabulator	1.25.1	py_0	conda-forge
text-unidecode	1.3	pyhd8ed1ab_1	conda-forge
tk	8.6.14	h39e8969_0
tomli	2.0.1	pyhd8ed1ab_0	conda-forge
types-python-dateutil	2.9.0.20240316	pyhd8ed1ab_0	conda-forge
typing-extensions	4.12.2	hd8ed1ab_0	conda-forge
typing_extensions	4.12.2	pyha770c72_0	conda-forge
tzdata	2024a	h0c530f3_0	conda-forge
unicodecsv	0.14.1	pyhd8ed1ab_2	conda-forge
urllib3	2.2.2	pyhd8ed1ab_1	conda-forge
wget	1.21.4	h91b91d3_1
wheel	0.43.0	pyhd8ed1ab_1	conda-forge
xlrd	2.0.1	pyhd8ed1ab_3	conda-forge
xz	5.4.6	h5eee18b_1
yaml	0.2.5	h7f98852_2	conda-forge
zipp	3.19.2	pyhd8ed1ab_0	conda-forge
zlib	1.2.13	h4ab18f5_6	conda-forge
zstandard	0.22.0	py39h81c9582_1	conda-forge
zstd	1.5.6	ha6fb4c9_0	conda-forge
```

If you do manage to get this and you can be bothered reporting back, I will update our conda recipe with specific versions to ensure this works for those with different compute environments in the future!

**Comment by sebbruchmann** [comment]
Hi @bananabenana ,

thanks so much - it worked and I managed to install it.
`bactabolize --version` gives me:

> bactabolize 1.0.3


I ran all your suggested individual lines and had to add just one additional one:
`micromamba install -y -c conda-forge blast==2.12.0`

And after that one I could run `micromamba install -y -c kelwyres -c bioconda -c conda-forge bactabolize==1.0.3`
And to confirm, yes all I have tried were in fresh environments (which I emptied before each new try).

Thanks so much, can't wait to try it.
Cheers,
Seb
